OADP 3144 - Installing OADP on an AWS STS cluster manually (#71534)
openshift/openshift-docs: modules/preparing-aws-sts-credentials-for-oadp.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/oadp-aws-sts/oadp-rosa-backing-up-applications.adoc

:_mod-docs-content-type: PROCEDURE
[id="preparing-aws-sts-credentials-for-oadp_{context}"]
= Preparing AWS STS credentials for OADP

An {aws-full} account must be prepared and configured to accept an {oadp-first} installation. Prepare the AWS credentials by following the proceeding steps.

.Procedure
. Define the `cluster_name` environment variable by running the following command:
+
[source,terminal]
----

> $ export CLUSTER_NAME=oadp-7105 <1>

Reviewer: @CarmiWisemon Maybe let's make this identical to the ROSA doc. 
```
export CLUSTER_NAME=my-cluster 
```
Replace the my-cluster with your AWS cluster name

----

> <1> The variable can be set to any value.

Reviewer 2: Should there be a + after this?

Author: No. The <1> is a call-out.

Author: I don't think we usually do that.

Reviewer 2: Not a +. My bad. I mean an empty line.

So...

Line 18: call out
Line 19: empty line
Line 20: . Retrieve all the details...

It may not matter, or it may not render properly in the Customer Portal without an empty line between the call out and the next step.

Author: @gwynnemonahan 
Thank you for the clarification.
I think we both got confused, because I was looking at a file with a similar name.
I added the space as you suggested.
Thanks!

> . Retrieve all the details of the `cluster_name` such as the `AWS_ACCOUNT_ID, OIDC_ENDPOINT` by running the following command:

Reviewer: ```suggestion
. Retrieve all the details of the `cluster` such as the `AWS_ACCOUNT_ID, OIDC_ENDPOINT` by running the following command:
```

+
[source,terminal]
----
$ export CLUSTER_VERSION=$(oc get clusterversion version -o jsonpath='{.status.desired.version}{"\n"}')

export AWS_CLUSTER_ID=$(oc get clusterversion version -o jsonpath='{.spec.clusterID}{"\n"}')

export OIDC_ENDPOINT=$(oc get authentication.config.openshift.io cluster -o jsonpath='{.spec.serviceAccountIssuer}' | sed 's|^https://||')

export REGION=$(oc get infrastructures cluster -o jsonpath='{.status.platformStatus.aws.region}' --allow-missing-template-keys=false || echo us-east-2)

> export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)

Reviewer 2: Should there be a + after this, or are 

export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
export ROLE_NAME="${CLUSTER_NAME}-openshift-oadp-aws-cloud-credentials"

supposed to be close together?

Author: I think it's supposed to be the way it is currently.

Reviewer 2: OK. Was curious as there is space between all the others except for the last 2, so was just curious. Thanks!

cat << EOF > ${SCRATCH}/policy.json
{
"Version": "2012-10-17",
"Statement": [
 {
   "Effect": "Allow",
   "Action": [
     "s3:CreateBucket",
     "s3:DeleteBucket",
     "s3:PutBucketTagging",
     "s3:GetBucketTagging",
     "s3:PutEncryptionConfiguration",
     "s3:GetEncryptionConfiguration",
     "s3:PutLifecycleConfiguration",
     "s3:GetLifecycleConfiguration",
     "s3:GetBucketLocation",
     "s3:ListBucket",
     "s3:GetObject",
     "s3:PutObject",
     "s3:DeleteObject",
     "s3:ListBucketMultipartUploads",
     "s3:AbortMultipartUpload",
     "s3:ListMultipartUploadParts",
     "ec2:DescribeSnapshots",
     "ec2:DescribeVolumes",
     "ec2:DescribeVolumeAttribute",
     "ec2:DescribeVolumesModifications",
     "ec2:DescribeVolumeStatus",
     "ec2:CreateTags",
     "ec2:CreateVolume",
     "ec2:CreateSnapshot",
     "ec2:DeleteSnapshot"
   ],
   "Resource": "*"
 }
]}
EOF

POLICY_ARN=$(aws iam create-policy --policy-name $POLICY_NAME \
--policy-document file:///${SCRATCH}/policy.json --query Policy.Arn \
--tags Key=openshift_version,Value=${CLUSTER_VERSION} Key=operator_namespace,Value=openshift-adp Key=operator_name,Value=oadp \
--output text) <1>
fi
----
+

> <1> `SCRATCH` is a name for a temporary directory created for the environment variables.

Reviewer: ```suggestion
<1> `SCRATCH` is a name for a temporary directory created for the storing the files.
```

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/oadp-aws-sts/oadp-aws-sts.adoc

:_mod-docs-content-type: PROCEDURE
[id="preparing-aws-sts-credentials-for-oadp_{context}"]
= Preparing AWS STS credentials for OADP

> An {aws-full} account must be prepared and configured to accept an {oadp-first} installation. Prepare the {aws-short} credentials by following the proceeding steps.

Reviewer 3: It might be simpler to say 

```suggestion
An {aws-full} account must be prepared and configured to accept an {oadp-first} installation. Prepare the {aws-short} credentials by using the following procedure.
```

.Procedure
. Define the `cluster_name` environment variable by running the following command:
+
[source,terminal]
----

> $ export CLUSTER_NAME= <AWS cluster name> <1>

Reviewer 3: See guidelines on replaceable values: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#command-syntax-for-replaceable-values 

```suggestion
$ export CLUSTER_NAME= <AWS_cluster_name> <1>
```

----
<1> The variable can be set to any value.

> . Retrieve all the details of the `cluster` such as the `AWS_ACCOUNT_ID, OIDC_ENDPOINT` by running the following command:

Reviewer 3: ```suggestion
. Retrieve all of the details of the `cluster` such as the `AWS_ACCOUNT_ID, OIDC_ENDPOINT` by running the following command:
```

+

> [source,terminal]

Reviewer 3: There is a space between all of these except the last two. Should you add a space there>?

Author: I'm think it's OK like this.

----
$ export CLUSTER_VERSION=$(oc get clusterversion version -o jsonpath='{.status.desired.version}{"\n"}')

export AWS_CLUSTER_ID=$(oc get clusterversion version -o jsonpath='{.spec.clusterID}{"\n"}')

export OIDC_ENDPOINT=$(oc get authentication.config.openshift.io cluster -o jsonpath='{.spec.serviceAccountIssuer}' | sed 's|^https://||')

export REGION=$(oc get infrastructures cluster -o jsonpath='{.status.platformStatus.aws.region}' --allow-missing-template-keys=false || echo us-east-2)

export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
export ROLE_NAME="${CLUSTER_NAME}-openshift-oadp-aws-cloud-credentials"
----

> . Create a temporary directory to store all the files  by running the following command:

Reviewer 3: extra space + addition of "of"

```suggestion
. Create a temporary directory to store all of the files by running the following command:
```

+
[source,terminal]
----
$ export SCRATCH="/tmp/${CLUSTER_NAME}/oadp"
mkdir -p ${SCRATCH}
----

> . Display all the gathered details by running the following command:

Reviewer 3: ```suggestion
. Display all of the gathered details by running the following command:
```

+
[source,terminal]
----
$ echo "Cluster ID: ${AWS_CLUSTER_ID}, Region: ${REGION}, OIDC Endpoint:
${OIDC_ENDPOINT}, AWS Account ID: ${AWS_ACCOUNT_ID}"
----
. On the {aws-short} account, create an IAM policy to allow access to {aws-short} S3:

.. Check to see if the policy exists by running the following commands:
+
[source,terminal]
----
$ export POLICY_NAME="OadpVer1" <1>
----

> <1> The variable can be set to any value.

Reviewer 3: What variable? This is just a question. Not sure if there's anything to fix

Author: OadpVer1
